static/app/components/globalAppStoreConnectUpdateAlert/updateAlert.tsx

```diff
@@ -1,67 +1,277 @@
+      return (
+        <ButtonClose
+          priority="link"
+          title={t('Dismiss')}
+          label={t('Dismiss')}
+          onClick={handleDismiss(sourceId)}
+          icon={<IconClose />}
+        />
+      );
+    }
+
+    return (
+      <Actions>
+        <Button priority="link" onClick={handleDismiss(sourceId)}>
+          {t('Dismiss')}
+        </Button>
+        |
+        <Button priority="link" to={projectSettingsLink}>
+          {t('Update session')}
+        </Button>
+      </Actions>
+    );
+  }
+
+  if (!project || !firstError || isDismissed) {
     return null;
   }
 
-  const notices = (
-    <Notices className={className}>
-      {Object.keys(appStoreConnectContext).map(key => {
-        const {updateAlertMessage} = appStoreConnectContext[key];
-        if (!updateAlertMessage) {
-          return null;
-        }
-
-        return (
-          <NoMarginBottomAlert key={key} type="warning" icon={<IconRefresh />}>
-            <AlertContent>{updateAlertMessage}</AlertContent>
-          </NoMarginBottomAlert>
-        );
-      })}
-    </Notices>
+  const projectSettingsLink = `/settings/${organization.slug}/projects/${project.slug}/debug-symbols/?customRepository=${firstError.id}`;
+
+  const notice = (
+    <Alert type="warning" icon={<IconRefresh />} className={className}>
+      <Content>
+        {renderMessage(firstError, projectSettingsLink)}
+        {renderActions(firstError.id, projectSettingsLink)}
+      </Content>
+    </Alert>
   );
 
-  return Wrapper ? <Wrapper>{notices}</Wrapper> : notices;
+  return Wrapper ? <Wrapper>{notice}</Wrapper> : notice;
 }
 
-export default UpdateAlert;
+export default withApi(UpdateAlert);
 
-const Notices = styled('div')`
+const Actions = styled('div')`
   display: grid;
-  grid-gap: ${space(2)};
-  margin-bottom: ${space(3)};
-`;
-
-const NoMarginBottomAlert = styled(Alert)`
-  margin-bottom: 0;
+  grid-template-columns: repeat(3, max-content);
+  grid-gap: ${space(1)};
+  align-items: center;
 `;
 
-const AlertContent = styled('div')`
+const Content = styled('div')`
   display: grid;
   grid-template-columns: 1fr max-content;
   grid-gap: ${space(1)};
 `;
+
+const ButtonClose = styled(Button)`
+  color: ${p => p.theme.textColor};
+  /* Give the button an explicit height so that it lines up with the icon */
+  height: 22px;
+`;
+
+/**
+ * Incredibly cursed multi prompt custom code zone
+ */
+
+type AppStoreConnectPromptsUpdateParams = {
+  /**
+   * The numeric organization ID as a string
+   */
+  organizationId: string;
+  /**
+   * The numeric project ID as a string
+   */
+  projectId: string;
+  /**
+   * The prompt feature name
+   */
+  feature: string;
+  /**
+   * The source ID
+   */
+  sourceId: string;
+  status: 'snoozed' | 'dismissed';
+};
+
+/**
+ * Update the status of an ASC prompt
+ */
+function promptsUpdate(api: Client, params: AppStoreConnectPromptsUpdateParams) {
+  return api.requestPromise('/app-store-connect-prompts-activity/', {
```

Comment:
shouldn't we only update the `prompts-activity` endpoint to support app-store-connect?